master/internal/templates/api_test.go

```diff
@@ -0,0 +1,306 @@
+//go:build integration
+
+package templates
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/test/testutils/apitest"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/templatev1"
+)
+
+func TestMain(m *testing.M) {
+	pgDB, err := db.ResolveTestPostgres()
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = db.MigrateTestPostgres(pgDB, "file://../../static/migrations", "up")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = etc.SetRootPath("../../static/srv")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestGetTemplate(t *testing.T) {
+	api := TemplateAPIServer{}
+	ctx := apitest.WithCredentials(context.Background())
+
+	t.Run("GetTemplate that does not exist", func(t *testing.T) {
```

Comment:
I acknowledge the soup of insanity. Maybe avoiding t.Run in favor of separate tests unless you are doing table driven tests is the best way to avoid that. I was using it for common setup, each `t.Run` tests are separately runnable (they don't depend on the order). I guess if using `t.Run` I would have a rule that you must be able to run each subtest, but, yeah, easier enforce by the first thing I said.

Reply:
this is the one thing i'm going to keep